src/restconfig/src/main/java/org/geoserver/catalog/rest/DataStoreResource.java

```diff
@@ -179,6 +180,16 @@ protected void configurePersister(XStreamPersister persister, DataFormat format)
         persister.setCallback( 
             new XStreamPersister.Callback() {
                 @Override
+                protected CatalogInfo getCatalogObject() {
```

Comment:
I added this method to the abstract superclass in XStreamPersister.Callback in the first place, so there was no previous implementation.
If you look in the Callback class, it is basically a bunch of empty methods that get overriden and are called during resource handling. This one is intended to get the proper catalog object for a given resource.

 If you look at the other overrriden methods below, you will see they folow the same coding pattern.